check_tv.py

```
@@ -386,15 +414,42 @@ def get_creds_saved():
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(APP_TOKEN_FILE, SCOPES)
+            creds = flow.run_local_server(port=0)
+
+        with open(USER_TOKEN_FILE, 'w') as token:
+            token.write(creds.to_json())
+    service = build('gmail', 'v1', credentials=creds)
     return service
-   
+  except HttpError as e:
```

Ruff (0.8.2)

Undefined name `HttpError`

(F821)